traversal/planner/PlannerVertex.java

```diff
@@ -41,43 +43,62 @@
 
     private final String varPrefix = "vertex_var_" + id() + "_";
     private final String conPrefix = "vertex_con_" + id() + "_";
-    private boolean isInitialisedVariables;
-    private boolean isInitialisedConstraints;
-    private double costNext;
-    double costLastRecorded;
+    private boolean didInitialiseVariables;
+    private boolean didInitialiseConstraints;
+    private boolean didInitialiseValues;
+
+    double cost;
+    double recordedCost;
+
     OptimiserVariable.Boolean varIsStartingVertex;
-    OptimiserVariable.Boolean varIsEndingVertex;
-    OptimiserVariable.Boolean varHasIncomingEdges;
-    OptimiserVariable.Boolean varHasOutgoingEdges;
+    OptimiserVariable.Boolean[] varOrderAssignment;
+    OptimiserVariable.Integer varOrderNumber;
 
     PlannerVertex(Identifier identifier, @Nullable GraphPlanner planner) {
         super(identifier);
         this.planner = planner;
-        isInitialisedVariables = false;
-        isInitialisedConstraints = false;
-        costLastRecorded = 0.01; // non-zero value for safe division
+        didInitialiseVariables = false;
+        didInitialiseConstraints = false;
+        didInitialiseValues = false;
+        recordedCost = 0.01; // non-zero value for safe division
```

Comment:
maybe that lets use remove the `max()` method that the `cost()` does internally?

Reply:
No, `cost` is set to zero so that we don't have to bother with `INIT_ZERO` until we access it. You'll notice that `cost` is never accessed directly, only through `cost()`. `max()` stays because of the way we determine whether the model is still applicable to the updated costs, which involves division between latest and recorded costs.

Reply:
It might read better if the accessor is called smth like `safeCost()` to emphasize the difference. It can definitely be confusing as is.